src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -451,24 +465,27 @@ trait WorkflowComponent {
     //Unmarshal all input resolutions for a single workflow. Assumes that everything in wfInputResolutionRecs has the same workflowId.
     def unmarshalOneWorkflowInputs(wfInputResolutionRecs: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])], workflowId: Long): Seq[SubmissionValidationValue] = {
 
-        //collect up the workflow resolution results by input
-        val resolutionsByInput = wfInputResolutionRecs.groupBy { case (resolution, attribute) => resolution.inputName }
-
-        //unmarshalAttributes will unmarshal multiple workflow attributes at once, but it expects all the attribute records to be real and not options.
-        //To get around this, we split by input, so that each input is successful (or not) individually.
-        val submissionValues = resolutionsByInput map { case (inputName, recTuples: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])]) =>
-          val attr = if( recTuples.forall { case (submissionRec, attrRecOpt) => attrRecOpt.isDefined } ) {
-            //all attributes are real
-            Some(
-              submissionAttributeQuery.unmarshalAttributes( recTuples map { case (rec, attrOpt) => ((workflowId, attrOpt.get), None) } )(workflowId)(inputName)
-            )
-          } else {
-            None
-          }
-          //assuming that the first elem has the error here
-          SubmissionValidationValue(attr, recTuples.head._1.errorText, inputName)
+      //collect up the workflow resolution results by input
+      val resolutionsByInput = wfInputResolutionRecs.groupBy { case (resolution, attribute) => resolution.inputName }
+
+      //unmarshalAttributes will unmarshal multiple workflow attributes at once, but it expects all the attribute records to be real and not options.
+      //To get around this, we split by input, so that each input is successful (or not) individually.
+      val submissionValues = resolutionsByInput map { case (inputName, recTuples: Seq[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])]) =>
+        val attr = if (recTuples.forall { case (submissionRec, attrRecOpt) => attrRecOpt.isDefined }) {
+          //all attributes are real
+          val attrRecsWithRefs = recTuples map { case (rec, Some(attrOpt)) => ((workflowId, attrOpt), None) }
+
+          // TODO fix this!
+          // this is an Input Resolution, not an Attribute Name.  Why are we storing it as an attribute?
+          val attrNameHack = AttributeName("default", inputName)
```

Comment:
AttributeName.defaultNamespace here too. I am also not convinced this is a hack.